packages/workspace/src/browser/workspace-commands.ts

```diff
@@ -368,19 +369,29 @@ export class WorkspaceCommandContribution implements CommandContribution {
         return new WorkspaceRootUriAwareCommandHandler(this.workspaceService, this.selectionService, handler);
     }
 
+    protected async validateFileRename(name: string, parent: FileStat, initialValue: string, allowOverwrite: boolean = false): Promise<string> {
+        if (OS.Type.Windows) {
```

Comment:
This will always be true, I believe you mean:

```ts
if (isWindows) { ... }
```

or

```ts
OS.type() === OS.Type.Windows
```

I'd go with `isWindows`.